File: Hash Table/3146. Permutation Difference between Two Strings/find_permutation_difference.go
package main

func findPermutationDifference(s string, t string) int {
	index := [26]int{}
	for i, ch := range s {
		index[ch-'a'] = i
	}
	result := 0
	for i, ch := range t {
		if index[ch-'a'] > i {
			index[ch-'a'] = index[ch-'a'] - i
		} else {
			index[ch-'a'] = i - index[ch-'a']
		}
		result += index[ch-'a']
	}
	return result
}

File: Hash Table/3146. Permutation Difference between Two Strings/find_permutation_difference.rs
impl Solution {
    pub fn find_permutation_difference(s: String, t: String) -> i32 {
        let mut index = [0; 26];
        for (i, ch) in s.chars().enumerate() {
            index[(ch as usize) - ('a' as usize)] = i;
        }
        let mut result = 0;
        for (i, ch) in t.chars().enumerate() {
            let mut diff = index[(ch as usize) - ('a' as usize)] as i32 - i as i32;
            if diff < 0 {
                diff = -diff;
            }
            result += diff;
        }
        result

    }
}